provider/identity_provider_resource.go

```diff
@@ -283,6 +310,22 @@ func (r *IdentityProviderResource) Create(ctx context.Context,
 			htpasswdBuilder.Password(state.HTPasswd.Password.Value)
 		}
 		builder.Htpasswd(htpasswdBuilder)
+    case state.Gitlab != nil:
+		builder.Type(cmv1.IdentityProviderType("GitlabIdentityProvider"))
```

Comment:
Please use the const from cmv1
`builder.Type(cmv1.IdentityProviderTypeGitlab)`

Reply:
You can refactor also the other cases :)

Reply:
Done and refactored other cases.